pkg/client/cache/delta_fifo.go

```diff
@@ -404,6 +416,16 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 	defer f.lock.Unlock()
 	for {
 		for len(f.queue) == 0 {
+			// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
```

Comment:
IMO, closing a fifo means we stop putting items in the queue, we wait for the queue to be drained completely, and all goroutines to exit, and then we have finished stopping. Please don't shut down with items still in the queue, that will make it hard to deterministically test.